vaas/vaas/router/forms.py

```diff
@@ -155,15 +154,16 @@ def __init__(self, *args, **kwargs):
             condition_domain = kwargs['instance'].src_domain
         super().__init__(*args, **kwargs)
         self.fields['priority'].initial = 250
+        self.fields['condition'] = ComplexRedirectConditionField()
         self.fields['condition'].widget.attrs.update({'condition_domain' : condition_domain})
         self.fields['destination'].widget.attrs.update({'class': 'form-control', 'placeholder': 'Destination path'})
         pretify_fields(self.fields.values())
 
-    def clean(self):
-        src_domain = self.data['condition_1']
+    def clean(self) -> None:
+        src_domain = DomainMapping.objects.get(pk=self.data['condition_1'])
         self.cleaned_data['src_domain'] = src_domain
 
-def pretify_fields(fields):
+def pretify_fields(fields) -> None:
```

Comment:
```
fields: Dict[str, ???]
```

Reply:
Here we have List instead of Dictionary.